DataSet.rst

```diff
@@ -1,276 +1,402 @@
+BAM records and labels a subdataset of the subreads as "Long Reads"::
+
+    <?xml version="1.0" encoding="utf-8"?>
+    <pbds:SubreadSet
+        xmlns="http://pacificbiosciences.com/PacBioDatasets.xsd"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xmlns:pbbase="http://pacificbiosciences.com/PacBioBaseDataModel.xsd"
+        xmlns:pbsample="http://pacificbiosciences.com/PacBioSampleInfo.xsd"
+        xmlns:pbmeta="http://pacificbiosciences.com/PacBioCollectionMetadata.xsd"
+        xmlns:pbds="http://pacificbiosciences.com/PacBioDatasets.xsd"
+        xsi:schemaLocation="http://pacificbiosciences.com/PacBioDataModel.xsd"
+        UniqueId="b095d0a3-94b8-4918-b3af-a3f81bbe519c"
+        TimeStampedName="subreadset_150304_231155"
+        MetaType="PacBio.DataSet.SubreadSet"
+        Name="DataSet_SubreadSet"
+        Tags=""
+        Version="4.0.0"
+        CreatedAt="2015-01-27T09:00:01">
+        <pbbase:ExternalResources>
+            <pbbase:ExternalResource
+                UniqueId="b095d0a3-94b8-4918-b3af-a3f81bbe5193"
+                TimeStampedName="subread_bam_150304_231155"
+                MetaType="PacBio.SubreadFile.SubreadBamFile"
+                ResourceId="m150404_101626_42267_s1_p0.1.subreads.bam">
+                <pbbase:FileIndices>
+                    <pbbase:FileIndex
+                        UniqueId="b095d0a3-94b8-4918-b3af-a3f81bbe5194"
+                        TimeStampedName="bam_index_150304_231155"
+                        MetaType="PacBio.Index.PacBioIndex"
+                        ResourceId="m150404_101626_42267_s1_p0.1.subreads.bam.pbi"/>
+                </pbbase:FileIndices>
+            </pbbase:ExternalResource>
+            <pbbase:ExternalResource
+                UniqueId="b095d0a3-94b8-4918-b3af-a3f81bbe5197"
+                TimeStampedName="subread_bam_150304_231155"
+                MetaType="PacBio.SubreadFile.SubreadBamFile"
+                ResourceId="m150404_101626_42267_s1_p0.2.subreads.bam">
+                <pbbase:FileIndices>
+                    <pbbase:FileIndex
+                        UniqueId="b096d0a3-94b8-4918-b3af-a3f81bbe5198"
+                        TimeStampedName="bam_index_150304_231155"
+                        MetaType="PacBio.Index.PacBioIndex"
+                        ResourceId="m150404_101626_42267_s1_p0.2.subreads.bam.pbi"/>
+                </pbbase:FileIndices>
+            </pbbase:ExternalResource>
+        </pbbase:ExternalResources>
+        <pbds:Filters>
+            <pbds:Filter>
+                <pbbase:Properties>
+                    <pbbase:Property Name="rq" Operator="gt" Value="0.80"/>
+                </pbbase:Properties>
+            </pbds:Filter>
+        </pbds:Filters>
+        <pbds:DataSetMetadata>
+            <pbbase:TotalLength>5000</pbbase:TotalLength>
+            <pbbase:NumRecords>500</pbbase:NumRecords>
+        </pbds:DataSetMetadata>
+        <pbds:DataSets>
+            <pbds:SubreadSet Name="Long Reads">
```

Comment:
It appears then that subDataSets can sometimes be valid datasets on their own, and other times simplified versions (This subDataSet is missing ExternalResources)? We should make this clear either way, as at least pbcore will construct DataSet objects from the relevant subDataSet XML tree as required, and it is nice to have those as first-class citizens without too much mapping of external resources.